tutu-backend/src/main/scala/fi/oph/tutu/backend/service/perusteluMuistio/PerusteluMuistioGenerator.scala

```diff
@@ -186,20 +191,128 @@ def haeTutkintokohtaisetTiedot(
                 }
               )
 
+          val suoritusvuodet = Seq(tutkinto.aloitusVuosi, tutkinto.paattymisVuosi).flatten
+            .mkString(" - ")
+
           Seq[String](
             s"Tutkinto ${tutkinto.jarjestys}:",
             s"  Tutkintotodistusotsikko: ${tutkinto.todistusOtsikko.getOrElse("-")}",
             s"  Nimi: ${tutkinto.nimi.getOrElse("-")}",
             s"  Pääaine tai erikoisala: ${tutkinto.paaaaineTaiErikoisala.getOrElse("-")}",
             s"  Korkeakoulun tai oppilaitoksen nimi: ${tutkinto.oppilaitos.getOrElse("-")}",
             s"  Korkeakoulun tai oppilaitoksen sijaintimaa: ${kielistettyMaakoodi.getOrElse("-")}",
-            s"  Todistuksen päivämäärä: ${tutkinto.todistuksenPaivamaara.getOrElse("-")}"
+            s"  Todistuksen päivämäärä: ${tutkinto.todistuksenPaivamaara.getOrElse("-")}",
+            s"  Suoritusvuodet: ${suoritusvuodet}",
+            s"  Ohjeellinen laajuus: ${tutkinto.ohjeellinenLaajuus.getOrElse("-")}",
+            s"  Tutkintoon sisältyi opinnäytetyö: ${tutkinto.opinnaytetyo.map(toKyllaEi).getOrElse("-")}",
+            s"  Tutkintoon sisältyi harjoittelu: ${tutkinto.harjoittelu.map(toKyllaEi).getOrElse("-")}",
+            s"  Lisätietoja opinnäytteisiin tai harjoitteluun liittyen: ${tutkinto.perustelunLisatietoja.getOrElse("-")}"
           ).mkString("\n")
         })
         .mkString("\n\n")
     })
 }
 
+def haeYleisetPerustelut(perusteluMaybe: Option[Perustelu]): Option[String] = {
+  if (perusteluMaybe == None) {
```

Comment:
```suggestion
 perusteluMaybe match {
    case None => None
    case Some(perustelu) => {...}
```
Meniskö tuo match-casella